### DISCUSSION TITLE
pushing commit

### DISCUSSION THREAD

**Discussion by barkatullahafridi** [community/community#57608] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Show & Tell

### Select Feature Area

- [X] Issues (The feature that allows you to track ideas, feedback, tasks, or bugs for work on GitHub. Not for any issue you are having).
- [ ] Projects (For planning and tracking work on GitHub - not any open source or other type of project).

### Body

i am facing  an issue!
when i push the  code to  GitHub the error message comes up for permission denied.
Note that i haven't created any user and  alos
Note that it works when i push from aws ubuntu server but when i push from my local the below message comes:

remote: Permission to barkatullahafridi/mavenproject.git denied to Asadullah551.
fatal: unable to access 'https://github.com/barkatullahafridi/mavenproject.git/': The requested URL returned error: 403

please if  anyone solve this issue

### Guidelines

- [X] I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @barkatullahafridi ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!

**Comment by MShazim** [comment]
> ### Select Topic Area
> Show & Tell
> 
> ### Select Feature Area
> * [x]  Issues (The feature that allows you to track ideas, feedback, tasks, or bugs for work on GitHub. Not for any issue you are having).
> * [ ]  Projects (For planning and tracking work on GitHub - not any open source or other type of project).
> 
> ### Body
> i am facing an issue! when i push the code to GitHub the error message comes up for permission denied. Note that i haven't created any user and alos Note that it works when i push from aws ubuntu server but when i push from my local the below message comes:
> 
> remote: Permission to barkatullahafridi/mavenproject.git denied to Asadullah551. fatal: unable to access 'https://github.com/barkatullahafridi/mavenproject.git/': The requested URL returned error: 403
> 
> please if anyone solve this issue
> 
> ### Guidelines
> * [x]  I have read the above statement and can confirm my post is relevant to the GitHub feature areas [Issues](https://docs.github.com/en/issues/tracking-your-work-with-issues/about-issues) and/or [Projects](https://docs.github.com/en/issues/planning-and-tracking-with-projects/learning-about-projects/about-projects).

Hey @barkatullahafridi ,
That error means you are using wrong credential or wrong repo:

1. Username needs to be the user, not the email
2. Ensure you provide the proper password. If you have 2FA enabled on GitHub, you need to make sure you create an Access Token with proper access permissions and use the token as password instead.
3. Repo format is: ` https://github.com/<user>/<repo>.git`

403 means GitHub is not granting you access so one or more of the 3 things above are the problem

**Reply by airtower-luna** [reply]
Passwords can't be used for Git operations with GitHub since a *very* long time, you need a token with or without 2FA. When creating the token make sure to select the **repo** scope (for repository access, like pushing) and if you use Actions also **workflow** (required to modify workflow files).
